### PR TITLE
Build WASM extensions

### PR DESCRIPTION
**Pull request by jwendell**

### CODE REVIEW

File: maistra/ci/pre-submit.sh [hunk 1]
```diff
@@ -18,7 +18,58 @@ export ARCH
...
+  --config=${ARCH} \
+  --local_ram_resources=12288 \
+  --local_cpu_resources=4 \
+  --jobs=4 \
+  --disk_cache=/bazel-cache \
+  //extensions:stats.wasm \
+  2>&1 | grep -v -E "${OUTPUT_TO_IGNORE}"
+
+bazel build \
+  --incompatible_linkopts_to_linklibs \
```
Comment (brian-avery) [inline maistra/ci/pre-submit.sh]: Can we simplify this? A lot of repeated code here and it needs to be extended as new extensions are added.

Comment (jwendell) [reply maistra/ci/pre-submit.sh]: don't spend time here. It's all experimental. Look at the branch.

Comment (brian-avery) [reply maistra/ci/pre-submit.sh]: +1. I'll approve. Thanks.

**Review by brian-avery — APPROVED** [review]

File: maistra/ci/pre-submit.sh [hunk 2]
```diff
@@ -18,7 +18,58 @@ export ARCH
 # Fix path to the vendor deps
 sed -i "s|=/work/|=$(pwd)/|" maistra/bazelrc-vendor
 
-# Build
+# Build WASM extensions first
+bazel build \
+  --incompatible_linkopts_to_linklibs \
```
Comment (dmitri-d) [inline maistra/ci/pre-submit.sh]: `incompatible_linkopts_to_linklibs` is used to correct the order of libraries on the command line under software collections environment (devtools-9 in our case). Don't need this with clang, unless it's being installed from a software collection.

Comment (dmitri-d) [reply maistra/ci/pre-submit.sh]: Ditto for other places where it's used here.

Comment (jwendell) [reply maistra/ci/pre-submit.sh]: We are building with gcc. We are using clang only to build the WASM extensions, which are not affected by this flags (they are a separate toolchain).

**Pull request merged**